client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const getRegExpFor = function( type, textToHighlight ) {
+	const expressions = {};
+
+	expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+	expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
+
+	return new RegExp( expressions[ type ], 'ig' );
+};
+
+const highlight = function( content, textToHighlight, type ) {
+	const matcher = getRegExpFor( type, textToHighlight ),
+		matches = matcher.exec( content );
+
+	if ( matches ) {
+		const highlights = [];
+		let highlighted = false;
+
+		for ( let i = 1, length = matches.length; i < length; i++ ) {
+			let item = matches[ i ];
+
+			if ( textToHighlight.toLowerCase() === item.toLowerCase() && ! highlighted ) {
+				item = <strong className="mentions__highlight" key={ i }>{ matches[ i ] }</strong>;
```

Comment:
The [`<mark>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark) is perhaps the more semantically correct choice here.